README.md

```diff
@@ -39,6 +39,152 @@ necessary for fetching and updating the packages.
   `nix/sources.nix` file that returns the sources as a Nix object.
 * [Show](#show): shows the packages' information.
 
+The next two sections cover [common use cases](#use-cases) and [full command
+description](#commands).
+
+### Use cases
+
+This covers three use cases:
+
+* [Bootstrapping a Nix project](#bootstrapping-a-nix-project).
+* [Tracking a different a different nixpkgs branch](#tracking-a-nixpkgs-branch).
+* [Fetching packages from custom URLs](#using-custom-urls).
+
+#### Bootstrapping a Nix project
+
+There is an `init` command that is useful when starting a new Nix project or
+when porting an existing Nix project to the niv versioning:
+
+``` shell
+$ niv init
+...
+$ tree
+.
+├── default.nix
+├── nix
+│   ├── default.nix
+│   ├── packages.nix
+│   ├── sources.json
+│   └── sources.nix
+└── shell.nix
+
+1 directory, 6 files
+```
+
+The files `default.nix`, `nix/default.nix` and `nix/packages.nix` are useful
+defaults for any Nix project. The file `nix/sources.json` is the file used by
+niv to store versions and is initialized with niv and nixpkgs:
+
+``` json
+{
+    "nixpkgs": {
+        "url": "https://github.com/NixOS/nixpkgs-channels/archive/....tar.gz",
+        "owner": "NixOS",
+        "branch": "nixos-18.09",
+        "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz",
+        "repo": "nixpkgs-channels",
+        "sha256": "...",
+        "description": "Nixpkgs/NixOS branches that track the Nixpkgs/NixOS channels",
+        "rev": "..."
+    },
+    "niv": {
+        "homepage": "",
```

Comment:
why is there an empty homepage here?

Reply:
Not sure, the GitHub API returning silly stuff